sockets/windows_socket.go

```diff
@@ -0,0 +1,18 @@
+// +build windows
+
+package sockets
+
+import (
+	"net"
+
+	"github.com/Microsoft/go-winio"
+)
+
+// NewWindowsSocket creates a Windows named pipe on the specified path.
+func NewWindowsSocket(addr string, pipeConfig *winio.PipeConfig) (net.Listener, error) {
```

Comment:
Follow the conventions of Go's `net` package. This should be called `Listen`. 

What is the purpose of adding this function? It looks like it could just be this:

```
var ListenWindowsSocket = winio.ListenPipe
```